File: AiHcmCms/Models/Notifications/Notification.cs
using AiHcmCms.Models.Users;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AiHcmCms.Models.Notifications
{
    public class Notification
    {
        public int Id { get; set; }
        public User Author { get; set; }
        public string Title { get; set; }
        public string Body { get; set; }
        public long Date { get; set; }
    }
}

File: client-app/src/actions/notification.js
import * as types from "./types";
import notification from "../api/notification";

const receiveNotifications = (notifications) => ({
  type: types.RECEIVE_NOTIFICATIONS,
  notifications,
});

export const getAllNotifications = () => (dispatch) => {
  var user = sessionStorage.getItem("token")
    ? JSON.parse(sessionStorage.getItem("token"))
    : [];
  notification.getNotifications(user.id).then((response) => {
    dispatch(receiveNotifications(response.data));
  });
};

export const markNotification = (idNotification, status) => (dispatch) => {
  var user = sessionStorage.getItem("token")
    ? JSON.parse(sessionStorage.getItem("token"))
    : [];
  notification.mark(user.id, idNotification, status).then(() => {
    dispatch({
      type: types.MARK_NOTIFICATION,
      idNotification: idNotification,
      status: status,
    });
  });
};

export const publicNotification = (title, body) => {
  var user = sessionStorage.getItem("token")
    ? JSON.parse(sessionStorage.getItem("token"))
    : [];
  var idAuthor = user.id;
  var date = Date.now();
  notification
    .insert(idAuthor, title, body, date)
    .then((response) => console.log(response));
};
